nautobot/core/views/__init__.py

```diff
@@ -132,6 +134,15 @@ def get(self, request, *args, **kwargs):
         return self.render_to_response(context)
 
 
+class MediaView(GenericView):
+    """
+    Wrap Django's serve() view to enforce LOGIN_REQUIRED for static media.
```

Comment:
This may be problematic actually, as the custom-branding files (`settings.BRANDING_FILEPATHS`) currently assume files stored in `settings.MEDIA_ROOT`, but those specific files **must** be accessible unauthenticated, for example to provide custom branding on the login page.

In the short term, we might have to carve out in this view an explicit exception for those files/paths in particular? In the longer term, I wonder if hosting them in `MEDIA_ROOT` was an incorrect design decision, and maybe (as a breaking change) we might move them to being expected in `STATIC_ROOT` instead?